src/main/java/net/dv8tion/jda/internal/entities/EntityBuilder.java

```diff
@@ -1059,6 +1059,29 @@ public Role createRole(GuildImpl guild, DataObject roleJson, long guildId)
             .setHoisted(roleJson.getBoolean("hoist"))
             .setColor(color == 0 ? Role.DEFAULT_COLOR_RAW : color)
             .setMentionable(roleJson.getBoolean("mentionable"));
+        if(roleJson.hasKey("tags"))
+        {
+            if(roleJson.hasKey("premium_subscriber"))
```

Comment:
these statements don't look like they would work, since they check for the key on the roleJson, instead of the sub-object "tags".
All the tag-specific role types could also be solved via RoleType#getTag() property and RoleType.fromTag(), and the default being NORMAL

Reply:
Replace the giant if statement and moved it to the RoleType#fromTags(), where the DataObject is passed and parsed from there. If has(), parse, if not, then just go with NORMAL.